### DISCUSSION TITLE
Why blocking a service doesn't block it's android app?

### DISCUSSION THREAD

**Discussion by seflm** [AdguardTeam/AdGuardHome#6919] [Q&A]
Hello,

when blocking certain services like Pinterest, Instagram, TikTok, and similar. The blocking works fine when trying to access the respective websites through a web browser, however, the respective Android apps still work uninterrupted. Can't figure out why. Do the apps have hardcoded IP addresses or use their own DNS services? Seems like when using such apps, there is no DNS request being made to Adguard Home according to logs.

Thanks guys

**Comment by amrmzr** [comment]
You have to block ports for that application then android app will also be blocked

**Reply by seflm** [reply]
Block the app ports using AdGuard? How do I do that? 

I thought AdGuard is just a DNS server, that it can block essentially everything as long as it asks for translating the domain name. I don't understand how app ports have anything to do with that.

Thanks!

**Reply by amrmzr** [reply]
Not using AdGuard Home, but in your router Advanced settings for specific IP/Mac address.

**Reply by seflm** [reply]
I am not sure I follow. What should be blocked on the router? Ports? Domains? 

Either way, the question here however was, how do the apps bypass the Adguard filters?

**Reply by amrmzr** [reply]
The Specific ports for that Website/App(e.g. TikTok) should be blocked from your main router that will block the website as well as its App.

The apps use different types of domains to access its content that's how they bypass I guess, unlike website(URL in address bar) just google to find the ports of www.tiktok.com and block it, it will block all access to that site and app.

**Reply by gohardhuddi** [reply]
I tried to block YouTube; it works on the laptop, but the Android app works well. then i add following in filter custom block list and youtube opens on android and even content like short videos loaded, but didn't play
||www.youtube.com^
||googlevideo.com^
||accounts.youtube.com^
